plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/unzipping/AlternatingLayerUnzipper.java

```diff
@@ -52,14 +52,44 @@ public class AlternatingLayerUnzipper implements ILayoutProcessor<LGraph> {
     @Override
     public void process (LGraph graph, IElkProgressMonitor progressMonitor) {
         
-        processLayerSplitProperty(graph);
-        
         int insertionLayerOffset = 1;
         List<Pair<Layer, Integer>> newLayers = new ArrayList<>();
         for (int i = 0; i < graph.getLayers().size(); i++) {
             
             int N = getLayerSplitProperty(graph.getLayers().get(i));
             boolean resetOnLongEdges = getResetOnLongEdgesProperty(graph.getLayers().get(i));
+            boolean minimizeEdgeLength = getminimizeEdgeLengthProperty(graph.getLayers().get(i));
+            
+            if (minimizeEdgeLength) {
+                // get maximum node width and average node height
+                double maxWidth = 0.0;
+                double averageHeight = 0.0;
+                
+                for (LNode node : graph.getLayers().get(i)) {
+                    maxWidth = Math.max(maxWidth, node.getSize().x);
+                    averageHeight += node.getSize().y;
+                }
+                
+                averageHeight /= graph.getLayers().get(i).getNodes().size();
+                
+                // add spacings for heuristic to get closer to zero spacing situation
+                maxWidth += Math.max(2 * graph.getProperty(LayeredOptions.SPACING_EDGE_NODE_BETWEEN_LAYERS),
+                        Math.max(graph.getLayers().get(i).getNodes().size() 
+                                * graph.getProperty(LayeredOptions.SPACING_EDGE_EDGE_BETWEEN_LAYERS),
+                                graph.getProperty(LayeredOptions.SPACING_NODE_NODE_BETWEEN_LAYERS)));
+                
+                averageHeight += Math.max(
+                        graph.getProperty(LayeredOptions.SPACING_NODE_NODE),
+                        graph.getProperty(LayeredOptions.SPACING_EDGE_NODE));
+                
+                // apply heuristic
+                // TODO: this heuristic is for the decision between N=1 and N=2, it shouldn't be used with N > 2
```

Comment:
Is this something you want to fix ?

Reply:
I'm not sure what the best way to approach this would be. In general I would say, normally one would not want to set N > 2. It might be possible to make the heuristic more advanced to support larger N, but I don't really see the value in it. So I would recommend not using the heuristic for larger N. Maybe just as a note in the melk documentation